### PR TITLE
wip: Feature/demo

### PR DESCRIPTION
**Pull request by jlunenburg**
@Rayman : I'll add some docstrings and if necessary some cleaning and then I'll remove the wip (hopefully within an hour). Furthermore, I'm not sure what the best way of handling all bundle stuff is.

I already fixed the namespacing but we might want to test this (don't have a RoboCup laptop at home)

@MatthijsBurgh : are you in the lab tomorrow? Could you doublecheck if there are no modified files on 'my' laptop in the robot-api folder. N.B.: this is *not* a ROS package so you can't use `roscd`, but it's located somewhere in between the other packages.

### CODE REVIEW

File: lib/body.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    this.action = new ROSLIB.ActionClient({
+      ros: ros,
+      serverName: 'body/joint_trajectory_action',
+      actionName: 'control_msgs/FollowJointTrajectoryAction',
+      timeout: 10,
+    });
+
+    // Get the skills parameters from the parameter server to extract the
+    // possible motions.
+    // ToDo: get rid of hardcoded "amigo"
```
Comment (jlunenburg) [inline lib/body.js]: We'd have to check this I guess: does it still work if we remove `/amigo/`. Rosbridge fixes the namespacing?

File: lib/body.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    this.right_gripper_action = new ROSLIB.ActionClient({
+      ros: ros,
+      serverName: 'right_arm/gripper/action',
+      actionName: 'tue_manipulation_msgs/GripperCommandAction',
+      timeout: 0.0,
+    });
+
+  }  // End of constructor
+
+  sendGoal(cmd) {
```
Comment (jlunenburg) [inline lib/body.js]: Should we replace this with sendJointGoal? Might be more convenient if we ever choose to add Cartesian goals as well

**Comment by MatthijsBurgh** [comment]
@jlunenburg I will be there. I will check your laptop

**Comment by MatthijsBurgh** [comment]
No modified files on athome12

File: lib/speech.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import {EventEmitter2} from 'eventemitter2';
+import {Message} from 'roslib';
```
Comment (Rayman) [inline lib/speech.js]: If you run `npx xo` you can check the style guide of this package, there are several small issues

File: lib/speech.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * Class exposes the interface to the text to speech 
+ * module of the robot
+ */
+class Speech extends EventEmitter2 {
```
Comment (Rayman) [inline lib/speech.js]: Speech doesn't publish events, so it doesn't need to inherit from EventEmitter2

File: lib/speech.js (hunk 3 above)
Comment (Rayman) [inline lib/speech.js]: If you run `npx xo` you can check the style guide of this package, there are several small issues

File: lib/body.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+  }  // End of constructor
+
+  /**
+   * Sends a body goal
+   * @param {object} cmd: cmd must contain two name, value pairs: 'body_part'
+   * (torso, left_arm or right_arm) identifies which part to use and 'configuration'
+   * the default configuration to which the part move.
+   */
+  sendGoal(cmd) {
```
Comment (Rayman) [inline lib/body.js]: If it has two parameters, you could just make sendGoal to have two parameters

File: lib/body.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+      // Send the goal with a default timeout of 10.0 seconds
+      goal.send(10.0);
+
+    }  // End of sendGoal
+
+  /**
+   * Sends a gripper goal
+   * @param {object} cmd: object containing 'side' ('left' or 'right') and 'direction' ('open' or 'close')
+   */
+  sendGripperGoal(cmd) {
```
Comment (Rayman) [inline lib/body.js]: Same here. Maybe we would mimic the robot_skills api:
```js
robot.body.sendLeftArmGoal('pose');
robot.body.openLeftGripper();
robot.body.closeRightGripper();
```

**Pull request merged**